docpipeline/local_generate.py

```diff
@@ -19,27 +19,23 @@
 from docpipeline import generate
 
 
-def process_local_blob(blob):
-    is_bucket = False
-    tmp_path, metadata, site_path = generate.build_and_format(blob, is_bucket)
-
-    # For local generation
-    output_path = blob.joinpath(metadata.name)
-    if output_path.exists():
-        log.info(f"deleting existing directory: {output_path}")
-        shutil.rmtree(output_path)
-    shutil.copytree(site_path, output_path, dirs_exist_ok=True)
-    shutil.rmtree(tmp_path)
-
-    log.success(f"Done with {metadata.name}!")
-
-
-def build_local_doc(input_path):
+def build_local_doc(input_path: pathlib.Path, output_dir: pathlib.Path = None):
```

Comment:
I think it'd be better to require callers provide `output_dir` as the behavior of what happens when they do not is a bit hidden in the code. A small comment could also clear this up.